tests/test_box_utils.py

```diff
@@ -0,0 +1,124 @@
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import unittest
+
+import torch
+from parameterized import parameterized
+
+from monai.data.box_utils import box_area, box_clip_to_image, box_convert_mode, box_convert_standard_mode
+from tests.utils import TEST_NDARRAYS
+
+# this script test box_convert_mode, box_convert_standard_mode, and box_area
+TESTS = []
+for p in TEST_NDARRAYS:
+    bbox = [[0, 0, 0, 0, 0, 0], [0, 1, 0, 2, 2, 3], [0, 1, 1, 2, 2, 3]]
+    image_size = [4, 4, 4]
+    TESTS.append(
+        [
+            {"bbox": p(bbox), "image_size": image_size, "mode": "xyzwhd"},
+            "xyzwhd",
+            [[0, 0, 0, 0, 0, 0], [0, 1, 0, 2, 2, 3], [0, 1, 1, 2, 2, 3]],
+            [0, 12, 12],
+        ]
+    )
+    TESTS.append(
+        [
+            {"bbox": p(bbox), "image_size": image_size, "mode": "xyzwhd"},
+            "xyzxyz",
+            [[0, 0, 0, 0, 0, 0], [0, 1, 0, 2, 3, 3], [0, 1, 1, 2, 3, 4]],
+            [0, 12, 12],
+        ]
+    )
+    TESTS.append(
+        [
+            {"bbox": p(bbox), "image_size": image_size, "mode": "xyzwhd"},
+            "xxyyzz",
+            [[0, 0, 0, 0, 0, 0], [0, 2, 1, 3, 0, 3], [0, 2, 1, 3, 1, 4]],
+            [0, 12, 12],
+        ]
+    )
+    TESTS.append(
+        [
+            {"bbox": p(bbox), "image_size": image_size, "mode": "xyzxyz"},
+            "xyzwhd",
+            [[0, 0, 0, 0, 0, 0], [0, 1, 0, 2, 1, 3], [0, 1, 1, 2, 1, 2]],
+            [0, 6, 4],
+        ]
+    )
+    TESTS.append(
+        [
+            {"bbox": p(bbox), "image_size": image_size, "mode": "xyzxyz"},
+            "xyzxyz",
+            [[0, 0, 0, 0, 0, 0], [0, 1, 0, 2, 2, 3], [0, 1, 1, 2, 2, 3]],
+            [0, 6, 4],
+        ]
+    )
+    TESTS.append(
+        [
+            {"bbox": p(bbox), "image_size": image_size, "mode": "xyzxyz"},
+            "xxyyzz",
+            [[0, 0, 0, 0, 0, 0], [0, 2, 1, 2, 0, 3], [0, 2, 1, 2, 1, 3]],
+            [0, 6, 4],
+        ]
+    )
+    TESTS.append(
+        [
+            {"bbox": p(bbox), "image_size": image_size, "mode": "xxyyzz"},
+            "xxyyzz",
+            [[0, 0, 0, 0, 0, 0], [0, 1, 0, 2, 2, 3], [0, 1, 1, 2, 2, 3]],
+            [0, 2, 1],
+        ]
+    )
+    TESTS.append(
+        [
+            {"bbox": p(bbox), "image_size": image_size, "mode": "xxyyzz"},
+            "xyzxyz",
+            [[0, 0, 0, 0, 0, 0], [0, 0, 2, 1, 2, 3], [0, 1, 2, 1, 2, 3]],
+            [0, 2, 1],
+        ]
+    )
+    TESTS.append(
+        [
+            {"bbox": p(bbox), "image_size": image_size, "mode": "xxyyzz"},
+            "xyzwhd",
+            [[0, 0, 0, 0, 0, 0], [0, 0, 2, 1, 2, 1], [0, 1, 2, 1, 1, 1]],
+            [0, 2, 1],
+        ]
+    )
+
+
+class TestCreateBoxList(unittest.TestCase):
```

Comment:
Nit: might be a good idea to write separate unit tests for the individual components to isolate errors from each other (this is a pure design decision though).